### DISCUSSION TITLE
blacksheep/requests/

### DISCUSSION THREAD

**Discussion by giscus** [Neoteroi/BlackSheep-Docs#43] [Q&A]
# blacksheep/requests/

The official documentation website of BlackSheep, a fast ASGI web framework for Python asyncio.

https://www.neoteroi.dev/blacksheep/requests/

**Comment by kostia-ivashchenko** [comment]
Is there a way to distinguish between `https://www.example.com/` and `https://www.example.com/?` requests?

**Reply by RobertoPrevato** [reply]
If you mean _distinguish_ in the sense of registering different request handlers for the same path but different query string ("?..."), that is not possible using a single router because routing is based on the URL path and HTTP method alone. You would need to check if a query string is present and apply the desired logic in the request handler.

**Reply by RobertoPrevato** [reply]
If instead you mean to distinguish exactly a web request with path "/" and a web request with path "/?", it is possible but you need to URL-encode the question mark sign, as it is otherwise reserved to describe a query string. In this second case, you can define different request handlers for "/" and "/%3F".
